### ISSUE TITLE
ProcessResources dependencies not declared

### ISSUE THREAD

**Issue by luca992** [icerockdev/moko-resources#375]
For each of my targets (macosX64, macosArm64, jvm, js, etc.) I get warnings like:

```
> Task :project-name:macosX64ProcessResources
Execution optimizations have been disabled for task ':project-name:macosX64ProcessResources' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 
  '/.../project-name/build/generated/moko/commonMain/res'. 
  Reason: Task ':project-name:macosX64ProcessResources' uses this output of task ':project-name:generateMRcommonMain' without declaring an explicit or implicit dependency. 
  This can lead to incorrect results being produced, depending on what order the tasks are executed. 
  Please refer to https://docs.gradle.org/7.5/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 
  '/.../project-name/build/generated/moko/macosX64Main/res'. 
  Reason: Task ':project-name:macosX64ProcessResources' uses this output of task ':kamel-tests:generateMRmacosX64Main' without declaring an explicit or implicit dependency. 
  This can lead to incorrect results being produced, depending on what order the tasks are executed. 
  Please refer to https://docs.gradle.org/7.5/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  ```
  
  To resolve the gradle warnings I have to declare for each target:
  
  ```
tasks.findByName("macosX64ProcessResources")!!.dependsOn("generateMRcommonMain")
tasks.findByName("macosX64ProcessResources")!!.dependsOn("generateMRmacosX64Main")
```

**Comment by mustafaozhan** [comment]
Confirming that having the same problem

**Comment by dalewking** [comment]
Here is workaround I use for iOS:

// Workaround for missing implicit dependencies
listOf("X64", "Arm64", "SimulatorArm64")
    .forEach { variant ->
        tasks.named("ios${variant}ProcessResources") {
            dependsOn("generateMRcommonMain")
            dependsOn("generateMRios${variant}Main")
        }
    }

**Comment by Alex009** [comment]
will be released in 0.21.2
